### PR TITLE
Some fixes for issues raised by cppcheck

### PR DESCRIPTION
**Pull request by notspiff**
Just some small stuff I quickly fixed while testing #12543

### CODE REVIEW

**Comment by Rechi** [comment]
@notspiff cpp files have to be updated as well, as currently function signature mismatches.

**Comment by notspiff** [comment]
that was stupid.. only ran the analysers.

**Comment by MartijnKaijser** [comment]
Jenkins build this please

File: xbmc/cores/AudioEngine/Utils/AEChannelInfo.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ class CAEChannelInfo {
 
 public:
   CAEChannelInfo();
-  CAEChannelInfo(const enum AEChannel* rhs);
+  explicit CAEChannelInfo(const enum AEChannel* rhs);
```
Comment (Rechi) [inline xbmc/cores/AudioEngine/Utils/AEChannelInfo.h]: this causes build issues for windows and android

Comment (notspiff) [inline xbmc/cores/AudioEngine/Utils/AEChannelInfo.h]: yes, i see. do you want to rely on implicit casts or should i change the code to be explicit?

Comment (Rechi) [inline xbmc/cores/AudioEngine/Utils/AEChannelInfo.h]: I think explicit way is preferred, but its your choice.

Comment (notspiff) [inline xbmc/cores/AudioEngine/Utils/AEChannelInfo.h]: if you ask me, you will always get the explicit code. i do not like implicit casts one bit.

**Comment by notspiff** [comment]
your CI is .. ehm "slightly annoying"

**Comment by Rechi** [comment]
>your CI is .. ehm "slightly annoying".
I know that already

jenkins build this please

**Comment by notspiff** [comment]
lol. one phone call, one skype session and coding on this at the same time was apparently too much for my brain.

**Comment by Rechi** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
Only osx fail due to code change

**Comment by Rechi** [comment]
jenkins build this please

**Comment by Rechi** [comment]
jenkins build this please

**Pull request merged**